### PR TITLE
Leaves_Ga-Young&Raisah

### PR DESCRIPTION
**Pull request by gyjin**
# Adagrams
Congratulations! You're submitting your assignment.

## Comprehension Questions

Feature | Feedback
--- | ---
What are the components that make up a method? | Method name, parameters, and block. If a return is not otherwise specified, returns the last statement executed.
What are the advantages of using git when collaboratively working on one code base? | Each person can individually fork from the code base, change the code, and suggest those changes back to the original code. Everyone can see the most updated code while also keeping multiple versions of their own code. 
What kind of relationship did you and your pair have with the unit tests? | At first, we didn't run the Rakefile after finishing wave-1 and manually tested our code. After hearing that other groups were running rake in between waves, we decided to run rake to find deficiencies in our code. While working on waves 2 - 4 we were aware that all parts of the test would not pass until the whole code was finished. The tests were helpful in determining which parts of our code were running correctly. 
Does your code use any methods from the `Enumerable` mixin? If so, where and why was it helpful? | We used has_key? to determine if a letters in the input word were part of the hand. We used .min to see if the input word used more letters than were available in the hand. We used .max_by and .min_by to compare lengths of the winning words in the event of a tie. These were "shortcuts" that made it easier to find specific data points without running nested loops.
What was one method you and your pair used to debug code? | We put print statements within adagrams.rb because the rake and wave files couldn't tell us specific problems with how our code was running. After finishing each wave, we used wave files and the rake file to test overall function. 
What are two discussion points that you and your pair discussed when giving/receiving feedback from each other that you would be willing to share? | We shared our specific ask/guess cultures and how we best approach problem solving with another person. We both identified with guess culture which meant that we communicated well but also wanted to improve on being more direct. We checked in halfway through our project time to make sure that our communication style was working.

### CODE REVIEW

File: lib/adagrams.rb [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    end
+  end
+  
+  # create randomized hand
+  shuffled_all_letters = all_letters.shuffle 
+  hand = []
+  
+  SIZE_OF_HAND.times do |i|
+    hand << shuffled_all_letters[i]
+  end
```
Comment (dHelmgren) [inline lib/adagrams.rb:54]: Solid process here!

File: lib/adagrams.rb [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+  # determines if letters in string are present in hand
+  input_as_array.each do |char|    
+    if hand_in_hash.has_key?(char) == true
+      hand_in_hash[char] -= 1
+    else
+      return false
+    end
+  end
+  
+  return hand_in_hash.values.min < 0 ? false : true
```
Comment (dHelmgren) [inline lib/adagrams.rb:84]: Nice! Clever! I dig it!

File: lib/adagrams.rb [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+    Z: 10
+  }
+  
+  # calculate score based on letters in word
+  score = 0
+  word_as_array.each do |char|
+    score += score_chart[char.to_sym]
+  end
+  
+  score += 8 if word.length > 6
```
Comment (dHelmgren) [inline lib/adagrams.rb:126]: Succinct!

**Comment by dHelmgren** [comment]
# Adagrams
## What We're Looking For

Feature | Feedback
--- | ---
**General** |
Answered comprehension questions | yes
Small commits with meaningful commit messages |yes
**Code Requirements** |
`draw_letters` method |
Uses appropriate data structure to store the letter distribution | yes
All tests for `draw_letters` pass | yes
`uses_available_letters?` method |
All tests for `uses_available_letters?` pass | yes
`score_word` method |
Uses appropriate data structure to store the letter scores | yes
All tests for `score_word` pass | yes
`highest_score_from` method |
Appropriately handles edge cases for tie-breaking logic | yes
All tests for `highest_score_from` pass | yes
**Overall** | Great code top to bottom, it's clear you really thought out the best tools you had for each step!
